src/config/artifacts.rs

```diff
@@ -0,0 +1,183 @@
+use crate::config::Config;
+use crate::errors::*;
+use crate::site::html::build_common_html;
+use axohtml::elements::{div, script, span};
+use axohtml::{html, text, unsafe_text};
+use cargo_dist_schema::{ArtifactKind, DistManifest};
+use serde::Deserialize;
+use std::fs::File;
+use std::io::Write;
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+enum ArtifactSystem {
+    Windows,
+    Windows64,
+    WindowsArm,
+
+    Mac,
+    MacPpc,
+    Mac32,
+    MacSilicon,
+
+    Linux,
+    LinuxUbuntu,
+    LinuxDebian,
+    LinuxMandriva,
+    LinuxRedhat,
+    LinuxFedora,
+    LinuxSuse,
+    LinuxGentoo,
+
+    Ios,
+    Android,
+
+    Freebsd,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Artifacts {
+    pub cargo_dist: bool,
+}
+
+fn get_kind_string(kind: &ArtifactKind) -> String {
+    match kind {
+        ArtifactKind::ExecutableZip => String::from("Executable Zip"),
+        ArtifactKind::Symbols => String::from("Symbols"),
+        ArtifactKind::Installer => String::from("Installer"),
+        _ => String::from("Unknown"),
+    }
+}
+
+fn create_download_link(config: &Config, name: &String) -> String {
+    if let (Some(repo), Some(version)) = (&config.repository, &config.version) {
+        format!("{}/releases/download/v{}/{}", repo, version, name)
+    } else {
+        String::new()
+    }
+}
+
+pub fn create_artifacts_tabs(config: &Config) -> Result<Option<Box<div<String>>>> {
```

Comment:
this is a big function may be valuable to try to break it into smaller ones if possible

Reply:
Done, created one for the install_hint